### ISSUE TITLE
Black produced different code on the second pass of the formatter

### ISSUE THREAD

**Issue by manolomartinez** [psf/black#2200]
Hello, just reporting this behavior, as requested by Black itself. This is the offending diff:

```diff
Mode(target_versions=set(), line_length=88, string_normalization=1, experimental_string_processing=False, is_pyi=False)
--- source
+++ first pass
@@ -543,13 +543,22 @@
 
     def one_codevector(self, new_conds, index):
         """
         Calculate the <inde> codevector.
         """
-        polynomial = new_conds[:, index, None, :] * np.array([-1, 3 * self.desig, -(3 * self.desig ** 2 +
-            (self.points[None, :, :, None] - self.desig)) ** 2, self.desig ** 3, self.desig *
-            (self.points[None, :, :, None] - self.desig) ** 2]).sum(0)
+        polynomial = new_conds[:, index, None, :] * np.array(
+            [
+                -1,
+                3 * self.desig,
+                -(
+                    (3 * self.desig ** 2 + (self.points[None, :, :, None] - self.desig))
+                    ** 2
+                ),
+                self.desig ** 3,
+                self.desig * (self.points[None, :, :, None] - self.desig) ** 2,
+            ]
+        ).sum(0)
         return np.roots(polynomial)
 
     def position_codevectors(self, new_conds, new_probs):
         """
         Calculate the new best codevector positions
--- first pass
+++ second pass
@@ -543,22 +543,28 @@
 
     def one_codevector(self, new_conds, index):
         """
         Calculate the <inde> codevector.
         """
-        polynomial = new_conds[:, index, None, :] * np.array(
-            [
-                -1,
-                3 * self.desig,
-                -(
-                    (3 * self.desig ** 2 + (self.points[None, :, :, None] - self.desig))
-                    ** 2
-                ),
-                self.desig ** 3,
-                self.desig * (self.points[None, :, :, None] - self.desig) ** 2,
-            ]
-        ).sum(0)
+        polynomial = (
+            new_conds[:, index, None, :]
+            * np.array(
+                [
+                    -1,
+                    3 * self.desig,
+                    -(
+                        (
+                            3 * self.desig ** 2
+                            + (self.points[None, :, :, None] - self.desig)
+                        )
+                        ** 2
+                    ),
+                    self.desig ** 3,
+                    self.desig * (self.points[None, :, :, None] - self.desig) ** 2,
+                ]
+            ).sum(0)
+        )
         return np.roots(polynomial)
 
     def position_codevectors(self, new_conds, new_probs):
         """
         Calculate the new best codevector positions
. . .

**Comment by xrisk** [comment]
I [can’t seem to reproduce this](https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AHdAPddAD2IimZxl1N_WlbvK5V-4TOSh0k8RCVC3-wtbV_iZoDllgLDozVo_Kqp-alzAnRd-iKoasJNEobAXkop0TqSCPehy43XU_YGmE7i2P6v0bXaD-LJcnzbGNNYmdbAaSmPV8HVrbqvahmRLJmTzJYLYosqX5nhZQkoedohHjGvR2MBE9C0nQPsgO51hgOMpf8pzVA0tK5xilyDMcT34MW9zwvqPZ9YJxMIAzcTYwvdpPDd2M_-HHBauaGUcKg3UPguY2b--hYor-Dgoh9pCMc6VhcR2Cgo8ZgpQhZvpMyvdlmHnciuy8ay3MllaEvMTRO-2Ku33bKOynwAAConeiFa-EX0AAGTAt4DAACjk6rmscRn-wIAAAAABFla) on the latest Black version or on master. What version of black are you on?

**Comment by manolomartinez** [comment]
I am running version 21.5b0-1, which is the latest one packaged for Arch linux. This is the output of `pacman -Si python-black`, in case it's useful:

Repository      : community
Name            : python-black
Version         : 21.5b0-1
Description     : Uncompromising Python code formatter
Architecture    : any
URL             : https://github.com/psf/black
Licenses        : MIT
Groups          : None
Provides        : None
Depends On      : python  python-appdirs  python-click  python-mypy_extensions  python-pathspec  python-regex  python-toml
Optional Deps   : python-aiohttp: for the blackd HTTP server
                  python-aiohttp-cors: for the blackd HTTP server
                  python-colorama: for colored diffs
                  python-typed-ast: to format Python 2 code
Conflicts With  : None
Replaces        : None
Download Size   : 208.32 KiB
Installed Size  : 1016.79 KiB
Packager        : Daniel M. Capella <polyzen@archlinux.org>
Build Date      : Wed May  5 00:43:23 2021
Validated By    : MD5 Sum  SHA-256 Sum  Signature

**Comment by manolomartinez** [comment]
... and the `vim` plugin up to commit `7b4ca`.

**Comment by JelleZijlstra** [comment]
This doesn't reproduce for me on the latest release. This most likely got fixed in the first 21.4 release, which fixed most stability issues. Most likely you have an old version of Black hanging around somewhere that gets picked up.

**Comment by manolomartinez** [comment]
I think you are right. Turns out that the vim plugin downloads its own Black, instead of using the system's. I had overlooked/forgotten this. Sorry for the noise.
